Sources/ITwinMobile/ITMApplication.swift

```diff
@@ -367,29 +379,42 @@ open class ITMApplication: NSObject, WKUIDelegate, WKNavigationDelegate {
     }
 
     /// Loads the iTwin Mobile web app backend.
+    ///
     /// Override this function in a subclass in order to add custom behavior.
     /// - Note: This function returns before the loading has completed.
     /// - Parameter allowInspectBackend: Whether or not to all debugging of the backend.
+    @MainActor
     open func loadBackend(_ allowInspectBackend: Bool) {
-        if backendLoaded {
+        if backendLoadStarted {
             return
         }
+        backendLoadStarted = true
         let backendUrl = getBackendUrl()
 
         authorizationClient = createAuthClient()
-        IModelJsHost.sharedInstance().loadBackend(
-            backendUrl,
-            withAuthClient: authorizationClient,
-            withInspect: allowInspectBackend
-        ) { _ in
-            // This callback gets called each time the app returns to the foreground. That is
-            // probably a bug in iModelJS, but this check avoids having that cause problems.
-            if !self.backendLoaded {
-                self.backendLoaded = true
-                self.backendLoadingDispatchGroup.leave()
+        Task {
+            // It's highly unlikely we could get here with backendLoadedContinuation nil,
+            // but if we do, we need to wait for it to be initialized before continuing, or
+            // nothing will work.
+            while backendLoadedContinuation == nil {
```

Comment:
Can we generalize the waitUntilReady function in ITMMessenger an re-use it here?

Reply:
Since it's watching a member variable waiting for it to become non-nil, I don't see how that could be done. If you know something about Swift that would allow me to do that that I don't know, I'd be happy to try. (In C/C++ it would require passing the address of the member variable into the function, but I don't think that's possible in Swift.)

Reply:
(I guess it could be done with a predicate, but I'm not sure that would be an improvement.)

Reply:
I'll do it with an isReady callback.

Reply:
Done.